File: flight_choice_prediction/v2/post/iOS SDK/FlightChoicePrediction.swift
import Amadeus

var amadeus: Amadeus = Amadeus(
    client_id: "YOUR_API_ID",
    client_secret: "YOUR_API_SECRET"
)


amadeus.shopping.flightOffersSearch.get(params:
   ["originLocationCode": "MAD",
    "destinationLocationCode": "BER",
    "departureDate": "2021-05-16",
    "returnDate": "2021-05-30",
    "adults": "2"],
    onCompletion: { result in

        switch result {
        case let .success(response):
            self.amadeus.shopping.flightOffers.prediction.post(body: response.result, onCompletion: {
                                                                   predictionResult in

                                                                   switch predictionResult {
                                                                   case let .success(response):
                                                                       print(response.data)
                                                                   case let .failure(error):
                                                                       fatalError(error.localizedDescription)
                                                                   }
                                                               })
        case let .failure(error):
            fatalError(error.localizedDescription)
        }
    })

File: flight_choice_prediction/v2/post/Python SDK/flight_choice_prediction.py
# Install the Python library from https://pypi.org/project/amadeus
from amadeus import Client, ResponseError

amadeus = Client(
    client_id='YOUR_AMADEUS_API_KEY',
    client_secret='YOUR_AMADEUS_API_SECRET'
)

try:
    '''
    Find the probability of the flight MAD to NYC to be chosen
    '''
    body = amadeus.shopping.flight_offers_search.get(originLocationCode='MAD',
                                                        destinationLocationCode='NYC',
                                                        departureDate='2020-11-01',
                                                        returnDate='2020-11-09',
                                                        adults=1).result
    response = amadeus.shopping.flight_offers.prediction.post(body)
    print(response.data)
except ResponseError as error:
    raise error
